### PR TITLE
Support custom nehuba instance

### PR DESCRIPTION
**Pull request by fsdavid**

### CODE REVIEW

File: src/components/ng-layer-tune/ng-layer-connector.ts [hunk 1]
```diff
@@ -20,10 +21,19 @@ async function getLayer(spec: TGetLayer) {
...
   } = spec
   const viewer = (viewerObj || (window as any).viewer)
+
+  let nehuba  
+
+  await new Promise((rs)=>setTimeout(() => {
+    nehuba = nehubaName && (window as any)[nehubaName]? (window as any)[nehubaName].ngviewer : viewer
+    rs('OK');
+  }, 1000)); 
+
```
Comment (xgui3783) [inline src/components/ng-layer-tune/ng-layer-connector.ts]:
```suggestion
  const viewer = (viewerObj || (window as any)[ viewerVariableName || 'viewer'])
```

File: src/components/ng-layer-tune/ng-layer-connector.ts [hunk 2]
```diff
@@ -20,10 +21,19 @@ async function getLayer(spec: TGetLayer) {
...
   return await retry(async () => {
-    const layerObj = viewer.layerManager.getLayerByName(layerName)
+    const layerObj = nehuba.layerManager.getLayerByName(layerName)
```
Comment (xgui3783) [inline src/components/ng-layer-tune/ng-layer-connector.ts]:
```suggestion
    const layerObj = viewer.layerManager.getLayerByName(layerName)
```

File: src/components/ng-layer-tune/ng-layer-connector.ts [hunk 3]
```diff
@@ -53,12 +63,12 @@ export class IntraFrameNglayerConnector implements NgLayerInterface {
   private ngLayer: PartialNgLayer
   public connected = false
 
-  constructor(private ngLayerName: string) {
+  constructor(private ngLayerName: string, private nehubaName?: string) {
```
Comment (xgui3783) [inline src/components/ng-layer-tune/ng-layer-connector.ts]:
```suggestion
  constructor(private ngLayerName: string, private viewerVariableName?: string) {
```

File: src/components/ng-layer-tune/ng-layer-tune.tsx [hunk 4]
```diff
@@ -44,6 +44,9 @@ export class NgLayerTune {
   @Prop()
   useNativeControl: boolean = false
 
+  @Prop()
+  nehubaName: string
+
```
Comment (xgui3783) [inline src/components/ng-layer-tune/ng-layer-tune.tsx]:
```suggestion
  @Prop()
  viewerVariableName: string

```

File: src/components/ng-layer-tune/readme.md [hunk 5]
```diff
@@ -16,6 +16,7 @@
 | `iFrameName`       | `i-frame-name`       |             | `string`      | `'ng-layer-tune'` |
 | `iframeLayerSpec`  | --                   |             | `NgLayerSpec` | `undefined`       |
 | `initialOpacity`   | `initial-opacity`    |             | `number`      | `null`            |
+| `nehubaName`       | `nehuba-name`        |             | `string`      | `undefined`       |
```
Comment (xgui3783) [inline src/components/ng-layer-tune/readme.md]:
```suggestion
| `viewerVariableName`       | `viewer-variable-name`        |             | `string`      | `undefined`       |
```

File: src/components/ng-layer-tune/ng-layer-tune.tsx [hunk 6]
```diff
@@ -163,7 +166,7 @@ export class NgLayerTune {
   private async coupleLayer(){
     if (!this.ngLayerName) return
     if (!this.useIframeCtrl) {
-      this.connector = new IntraFrameNglayerConnector(this.ngLayerName)
+      this.connector = new IntraFrameNglayerConnector(this.ngLayerName, this.nehubaName)
```
Comment (xgui3783) [inline src/components/ng-layer-tune/ng-layer-tune.tsx]:
```suggestion
      this.connector = new IntraFrameNglayerConnector(this.ngLayerName, this.viewerVariableName)
```

File: src/components/ng-layer-tune/ng-layer-connector.ts [hunk 7]
```diff
@@ -53,12 +63,12 @@ export class IntraFrameNglayerConnector implements NgLayerInterface {
...
 
   }
   async init(){
-
-    const layerObj = await getLayer({ layerName: this.ngLayerName })
+    // const layerObj = await getLayer({ layerName: this.ngLayerName, nehubaName: this.nehubaName || null });
+    const layerObj = await getLayer({ layerName: this.ngLayerName, nehubaName: this.nehubaName || null });
```
Comment (xgui3783) [inline src/components/ng-layer-tune/ng-layer-connector.ts]:
```suggestion
    const layerObj = await getLayer({ layerName: this.ngLayerName, viewerVariableName: this.viewerVariableName });
```

**Review by xgui3783 — CHANGES_REQUESTED** [review]
There should not be any references to nehuba. This project and nehuba are independent.
Otherwise, looking good.

File: src/components/ng-layer-tune/ng-layer-connector.ts (hunk 1 above)
Comment (fsdavid) [reply src/components/ng-layer-tune/ng-layer-connector.ts]: for a custom nehuba name, it requires adding `ngviewer` as one more step before access to viewer

Comment (xgui3783) [reply src/components/ng-layer-tune/ng-layer-connector.ts]: You can assign .ngviewer to a different variable on the global space

I would not like to closely couple ng layer tune with nehuba

**Comment by fsdavid** [comment]
@xgui3783 Changes are made and it's ready to be merged and released

File: src/components/ng-layer-tune/ng-layer-connector.ts [hunk 8]
```diff
@@ -20,8 +21,20 @@ async function getLayer(spec: TGetLayer) {
...
-  const viewer = (viewerObj || (window as any).viewer)
+  // const viewer = (viewerObj || (window as any).viewer)
+
+  // let nehuba  
+
+  // await new Promise((rs)=>setTimeout(() => {
+  //   nehuba = nehubaName && (window as any)[nehubaName] && (window as any)[nehubaName].ngviewer?
+  //     (window as any)[nehubaName].ngviewer : viewer
+  //   rs('OK');
+  // }, 1000));
```
Comment (xgui3783) [inline src/components/ng-layer-tune/ng-layer-connector.ts]: Please remove

File: src/components/ng-layer-tune/ng-layer-connector.ts [hunk 9]
```diff
@@ -53,12 +66,12 @@ export class IntraFrameNglayerConnector implements NgLayerInterface {
...
   public connected = false
 
-  constructor(private ngLayerName: string) {
+  constructor(private ngLayerName: string, private viewerVariableName?: string) {
 
   }
   async init(){
-
-    const layerObj = await getLayer({ layerName: this.ngLayerName })
+    // const layerObj = await getLayer({ layerName: this.ngLayerName, nehubaName: this.nehubaName || null });
```
Comment (xgui3783) [inline src/components/ng-layer-tune/ng-layer-connector.ts]: please remove

**Review by xgui3783 — CHANGES_REQUESTED** [review]

**Comment by xgui3783** [comment]
ping @fsdavid

**Comment by fsdavid** [comment]
@xgui3783 Done. Shame on me :)

File: package-lock.json
Comment (xgui3783) [inline package-lock.json:1]: did you intend to remove package-lock.json? @fsdavid

Comment (fsdavid) [reply package-lock.json:1]: Actually not. It's restored

**Pull request merged**